### ISSUE TITLE
Versioning strategy

### ISSUE THREAD

**Issue by with-heart** [arackaf/customize-cra#202]
The longer this project exists, the more difficult it becomes to keep the project in sync with existing `customize-cra` versions. 

As an example, `create-react-app@^3.3.0` now uses `css-loader@^3.0.0` which requires a breaking change in the implementation of style loaders because `modules` is now an object rather than a boolean flag.

How can we best manage versions to support these diverging configuration requirements while allowing users to easily know which version to use for their project?

My initial thought is to stop using `semver` and mirror the `customize-cra` versions to the equivalent `create-react-app` version, e.g. the version of `customize-cra` that supports the new `css-loader` changes in `create-react-app@3.3.0` would be `customize-cra@3.3.0`.

I could use some feedback on this though. I'm not sure how to best manage this type of complexity when we're entirely dependent on an upstream project.

cc: @FezVrasta

**Comment by FezVrasta** [comment]
With your proposed solution, how would you handle new features/fixes releases that target the same CRA version? Would you always use the minor version for this? (0.0.x)

**Comment by with-heart** [comment]
Yeah, I think that'd be the plan. X.Y.Z where X & Y mirror `create-react-app` and then Z is for our own versioning. I've seen this pattern in the wild before, I'll see if I can find some examples.
